fuzzing/fuzz_direct_parse.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2024 Mikhail Khachayants (mkhachaiants@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/json
+//
+
+#include <boost/json/parse_into.hpp>
+#include <boost/describe.hpp>
+#include <map>
+
+#ifndef BOOST_NO_CXX17_HDR_VARIANT
+# include <variant>
+# define IF_CXX17_HDR_VARIANT(...) __VA_ARGS__
+#else
+# define IF_CXX17_HDR_VARIANT(...)
+#endif // BOOST_NO_CXX17_HDR_VARIANT
+
+#ifndef BOOST_NO_CXX17_HDR_OPTIONAL
+# include <optional>
+# define IF_CXX17_HDR_OPTIONAL(...) __VA_ARGS__
+#else
+# define IF_CXX17_HDR_OPTIONAL(...)
+#endif // BOOST_NO_CXX17_HDR_OPTIONAL
+
+using namespace boost::json;
+
+struct Object
+{
+    bool b;
+    float f;
+    double d;
+    std::int64_t i64;
+    std::uint64_t u64;
+    std::string s;
+    std::vector<bool> v1;
+    std::vector<std::int64_t> v2;
+    std::vector<std::uint64_t> v3;
+    std::array<bool, 3> a1;
+    std::array<std::int64_t, 3> a2;
+    std::array<std::uint64_t, 3> a3;
+    std::map<std::string, std::int64_t> m1;
+    std::map<std::string, std::string> m2;
+    std::map<std::string, double> m3;
+    std::tuple<bool, std::uint64_t, std::int64_t, double, std::string> t1;
+    std::tuple<std::array<std::string, 3>, std::array<double, 3>, std::nullptr_t> t2;
+    std::tuple<std::vector<std::string>, std::vector<double>> t3;
+
+#ifndef BOOST_NO_CXX17_HDR_VARIANT
+    std::variant<bool, std::uint64_t, std::int64_t, double, std::string> v;
```

Comment:
Please, replace with `boost::variant2::variant`. It will reduce the amount of macro switching necessary.

Reply:
Done